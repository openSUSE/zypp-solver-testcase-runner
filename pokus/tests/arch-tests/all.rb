#!/usr/bin/ruby


# Generate YUM repository containing a package for all architectures.  Check
# for architectures that the correct architectures are loaded into the pool.


require 'zypptools/lib/arch'
require 'zypptools/lib/package'
require 'zypptools/lib/pattern'
require 'zypptools/lib/selection'
require 'zypptools/lib/product'
require 'zypptools/output/output'
include ZYppTools

require '../../src/testsuite'


packages = Array.new()

Arch.all_archs.each do |arch|
    pkg = Package.new("test")
    pkg.arch = arch
    pkg.version = "1.2"
    pkg.release = 1
    pkg.summary = "A pure test package"
    packages.push(pkg)
end


path = Testsuite.write_repo(:yum, packages)


Arch.all_archs.map{|x|x.to_s}.sort.each do |arch|

    x = Arch.new(arch)

    puts "#{x.arch} -> #{x.compat.map{|y|y.to_s}.sort.join(' ')}"

    Testsuite.set_arch(x.arch)

    t = Array.new()

    pool = Testsuite.read_repo("file://" + path)
    Testsuite::haha2(pool).each do |res|
        puts "#{res.kind_to_s} #{res.name} #{res.edition.to_s} #{res.arch.to_s}"
        t.push(res.arch.to_s)
    end

    if not x.compat.map{|y|y.to_s}.sort.eql?(t.sort())
        puts "Error: archs don't match"
        exit 1
    end

    puts

end

