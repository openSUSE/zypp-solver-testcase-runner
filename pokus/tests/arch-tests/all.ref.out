athlon -> athlon i386 i486 i586 i686 noarch
package test 1.2-1 noarch
package test 1.2-1 i386
package test 1.2-1 i486
package test 1.2-1 i586
package test 1.2-1 i686
package test 1.2-1 athlon

i386 -> i386 noarch
package test 1.2-1 noarch
package test 1.2-1 i386

i486 -> i386 i486 noarch
package test 1.2-1 noarch
package test 1.2-1 i386
package test 1.2-1 i486

i586 -> i386 i486 i586 noarch
package test 1.2-1 noarch
package test 1.2-1 i386
package test 1.2-1 i486
package test 1.2-1 i586

i686 -> i386 i486 i586 i686 noarch
package test 1.2-1 noarch
package test 1.2-1 i386
package test 1.2-1 i486
package test 1.2-1 i586
package test 1.2-1 i686

ia64 -> i386 i486 i586 i686 ia64 noarch
package test 1.2-1 noarch
package test 1.2-1 i386
package test 1.2-1 i486
package test 1.2-1 i586
package test 1.2-1 i686
package test 1.2-1 ia64

noarch -> noarch
package test 1.2-1 noarch

ppc -> noarch ppc
package test 1.2-1 noarch
package test 1.2-1 ppc

ppc64 -> noarch ppc ppc64
package test 1.2-1 noarch
package test 1.2-1 ppc
package test 1.2-1 ppc64

s390 -> noarch s390
package test 1.2-1 noarch
package test 1.2-1 s390

s390x -> noarch s390 s390x
package test 1.2-1 noarch
package test 1.2-1 s390
package test 1.2-1 s390x

x86_64 -> athlon i386 i486 i586 i686 noarch x86_64
package test 1.2-1 noarch
package test 1.2-1 i386
package test 1.2-1 i486
package test 1.2-1 i586
package test 1.2-1 i686
package test 1.2-1 athlon
package test 1.2-1 x86_64

